src/components/plots/barPlot.ts

```diff
@@ -560,17 +558,16 @@ export module Plots {
     }
 
     protected _getDataToDraw() {
-      var datasets: D3.Map<any[]> = d3.map();
+      var dataToDraw = new Utils.Map<Dataset, any[]>();
```

Comment:
Looking at this, it seems like a better implementation might be to return a filter method that's applied to the `Dataset.data()` wherever that winds up being used.